contract/src/msgs/data_requests/sudo/remove_requests.rs

```diff
@@ -1,14 +1,147 @@
+                        "type": to_json_string(&message.type_)?,
+                        "kind": to_json_string(&message.kind)?,
+                    }))?,
+                );
+            }
+            DistributionKind::ExecutorReward(distribution_executor_reward) => {
+                let public_key = PublicKey::from_hex_str(&distribution_executor_reward.identity)?;
+                let mut staker = STAKERS.get_staker(deps.storage, &public_key)?;
+
+                let minimum_stake = STAKING_CONFIG.load(deps.storage)?.minimum_stake_to_register;
+                let remaining_reward = if staker.tokens_staked < minimum_stake {
+                    // top the staker up to minimum stake from the amount in the reward & escrow
+                    let top_up = minimum_stake.checked_sub(staker.tokens_staked)?;
+                    let top_up = top_up.min(distribution_executor_reward.amount);
+                    staker.tokens_staked += top_up;
+                    dr_escrow.amount = dr_escrow.amount.checked_sub(top_up)?;
+                    distribution_executor_reward.amount.checked_sub(top_up)?
+                } else {
+                    distribution_executor_reward.amount
+                };
+
+                // send remaining reward to the staker pending withdrawal
+                staker.tokens_pending_withdrawal += remaining_reward;
+                dr_escrow.amount = dr_escrow.amount.checked_sub(remaining_reward)?;
+                STAKERS.update(deps.storage, public_key, &staker)?;
+
+                event = event.add_attribute(
+                    "executor_reward",
+                    to_json_string(&json!({
+                        "amount": distribution_executor_reward.amount,
+                        "identity": distribution_executor_reward.identity,
+                        "type": to_json_string(&message.type_)?,
+                        "kind": to_json_string(&message.kind)?,
+                    }))?,
+                );
+            }
+            DistributionKind::Send(distribution_send) => {
+                dr_escrow.amount = dr_escrow.amount.checked_sub(distribution_send.amount)?;
+                bank_messages.push(BankMsg::Send {
+                    to_address: Addr::from_vec(distribution_send.to.to_vec())?.to_string(),
+                    amount:     vec![amount_to_tokens(distribution_send.amount, token)],
+                });
+                event = event.add_attribute(
+                    "send",
+                    to_json_string(&json!({
+                        "amount": distribution_send.amount,
+                        "to": distribution_send.to,
+                        "type": to_json_string(&message.type_)?,
+                        "kind": to_json_string(&message.kind)?,
+                    }))?,
+                );
+            }
+        }
+    }
+
+    if !dr_escrow.amount.is_zero() {
+        bank_messages.push(BankMsg::Send {
+            to_address: dr_escrow.staker.to_string(),
```

Comment:
This should be `poster`
```suggestion
            to_address: dr_escrow.poster.to_string(),
```